src/tests/dbus-tests/test_zram.py

```diff
@@ -0,0 +1,181 @@
+
+        # /usr/local/lib/zram.conf.d
+        if os.path.exists(ZRAMCONFDIR):
+            for fname in os.listdir(ZRAMCONFDIR):
+                cls.conf[ZRAMCONFDIR][fname] = cls.read_file(ZRAMCONFDIR + '/' + fname)
+
+    @classmethod
+    def _restore_conf_files(cls):
+        # /usr/lib/modprobe.d/zram.conf
+        if cls.conf[MODPROBECONF]:
+            cls.write_file(MODPROBECONF, cls.conf[MODPROBECONF])
+
+        # /usr/lib/modules-load.d/zram.conf
+        if cls.conf[MODLOADCONF]:
+            cls.write_file(MODLOADCONF, cls.conf[MODLOADCONF])
+
+        # /usr/local/lib/zram.conf.d
+        for fname in cls.conf[ZRAMCONFDIR].keys():
+            cls.write_file(ZRAMCONFDIR + '/' + fname, cls.conf[ZRAMCONFDIR][fname])
+
+    @classmethod
+    def setUpClass(cls):
+        storagedtestcase.StoragedTestCase.setUpClass()
+        cls.ensure_modules_loaded()
+
+        cls._save_conf_files()
+
+    @classmethod
+    def tearDownClass(cls):
+        storagedtestcase.StoragedTestCase.tearDownClass()
+
+        cls._restore_conf_files()
+
+        cls.run_command('rmmod zram')
+
+    def _get_zrams(self):
+        time.sleep(0.5)
+        udisks = self.get_object('')
+        objects = udisks.GetManagedObjects(dbus_interface='org.freedesktop.DBus.ObjectManager')
+
+        return [path for path in objects.keys() if re.match(r'.*/block_devices/zram[0-9]$', path)]
+
+    def _swapoff(self, swap):
+        self.run_command('swapoff %s' % swap)
+
+    def test_create_destroy(self):
+        manager = self.get_object('/Manager')
+        manager.CreateDevices([10 * 1024**2, 10 * 1024**2], [1, 2], self.no_options,
+                              dbus_interface=self.iface_prefix + '.Manager.ZRAM')
+        time.sleep(3)
+
+        zrams = self._get_zrams()
+        self.assertEqual(len(zrams), 2)
+
+        # zram devices properties
+        for path in zrams:
+            zram = self.bus.get_object(self.iface_prefix, path)
+            self.assertIsNotNone(zram)
+
+            zram_name = path.split('/')[-1]
+            self.assertTrue(os.path.exists('/dev/%s' % zram_name))
+
+            sys_size = self.read_file('/sys/block/%s/disksize' % zram_name).strip()
+            dbus_size = self.get_property(zram, '.Block.ZRAM', 'disksize')
+            dbus_size.assertEqual(int(sys_size))
+
+            sys_alg = self.read_file('/sys/block/%s/comp_algorithm' % zram_name).strip()
+            dbus_alg = self.get_property(zram, '.Block.ZRAM', 'comp_algorithm')
+            dbus_alg.assertEqual(sys_alg)
+
+            sys_streams = self.read_file('/sys/block/%s/max_comp_streams' % zram_name).strip()
+            dbus_streams = self.get_property(zram, '.Block.ZRAM', 'max_comp_streams')
+            dbus_streams.assertEqual(int(sys_streams))
+
+        # destroy zrams
+        manager.DestroyDevices(self.no_options,
+                               dbus_interface=self.iface_prefix + '.Manager.ZRAM')
```

Comment:
What if manager.DestroyDevices fails? Will that be a problem for the next test run? If so it might be a good idea to ensure zrams are removed properly.